### PR TITLE
Rest API for the RP Microsoft.AppPlatform

### PR DESCRIPTION
**Pull request by bowen5**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by iRainbow5** [comment]
Corresponding pull request has been signed off https://github.com/Azure/azure-rest-api-specs-pr/pull/894 in azure-rest-api-specs private repo.

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2019-05-01-preview/appplatform.json [hunk 1]
```diff
@@ -0,0 +1,2818 @@
...
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AppPlatform/locations/{location}/checkNameAvailability": {
+      "post": {
+        "tags": [
+          "AppPlatform"
+        ],
+        "description": "Checks that the resource name is valid and is not already in use.",
+        "operationId": "CheckNameAvailability",
```
Comment (ArcturusZhang) [inline specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2019-05-01-preview/appplatform.json]: I recommend you to add a noun prefix to this operationId as Joel does too, for instance `Operations_CheckNameAvailability`.
Autorest will depend on this convention, and I am not sure what will happen if this convention is not followed.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Please see the comments

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2019-05-01-preview/appplatform.json (hunk 1 above)
Comment (iRainbow5) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2019-05-01-preview/appplatform.json]: Resolved. Thanks.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
@iRainbow5 Please let me know when this is ready to merge.

**Pull request merged**